### PR TITLE
Implement Iterator on ImmutableArray

### PR DESCRIPTION
**Pull request by EHER**
In order to have ImmutableArray as close as possible as a classic array Iterator should be implemented.

### CODE REVIEW

**Review by airtonzanon — APPROVED** [review]
LGTM

File: test/unit/ImmutableArrayTest.php [hunk 1]
```diff
@@ -75,4 +75,19 @@ public function it_should_prevent_to_unset_an_item(): void
...
+    /**
+     * @test
+     */
+    public function it_should_be_iterable(): void
+    {
+        $immutableArray = new ImmutableArray(['a', 'b', 'c']);
+        $expectedArray = [];
+
+        foreach ($immutableArray as $key => $value) {
+            $expectedArray[$key] = $value;
```
Comment (malukenho) [inline test/unit/ImmutableArrayTest.php:88]: this is `$actual`, not `$expected`

File: test/unit/ImmutableArrayTest.php [hunk 2]
```diff
@@ -75,4 +75,19 @@ public function it_should_prevent_to_unset_an_item(): void
 
         unset($immutableArray[0]);
     }
+
+    /**
+     * @test
+     */
+    public function it_should_be_iterable(): void
+    {
+        $immutableArray = new ImmutableArray(['a', 'b', 'c']);
```
Comment (malukenho) [inline test/unit/ImmutableArrayTest.php:84]:
```php
$expected = ['a', 'b', 'c'];
$immutableArray = new ImmutableArray($expected);

// ...

self::assertSame($actual, $expected);
```

**Review by malukenho — CHANGES_REQUESTED** [review]

File: src/ImmutableArray.php [hunk 3]
```diff
@@ -5,17 +5,20 @@
...
-class ImmutableArray implements ArrayAccess
+class ImmutableArray implements ArrayAccess, Iterator
 {
     private array $data;
+    private int $position;
 
     public function __construct(array $data)
     {
         $this->data = $data;
+        $this->position = 0;
```
Comment (malukenho) [inline src/ImmutableArray.php:21]: you can do it on the initialization itself `private int $position = 0;`

**Comment by malukenho** [comment]
As talked offline, it would be nice to have a test with `array_map` for example.

**Review by malukenho — APPROVED** [review]

**Pull request closed without merging**